core/src/main/java/com/sequenceiq/cloudbreak/service/events/DefaultCloudbreakEventService.java

```diff
@@ -121,18 +121,27 @@ public void fireCloudbreakInstanceGroupEvent(Long stackId, String eventType, Str
     public List<StructuredNotificationEvent> cloudbreakEventsForStack(Long stackId) {
         List<StructuredNotificationEvent> events = new ArrayList<>();
         if (stackId != null) {
-            events = structuredEventService.getEventsWithTypeAndResourceId(StructuredNotificationEvent.class, "stacks", stackId);
+            events = structuredEventService.getEventsWithTypeAndResourceId(StructuredNotificationEvent.class,
+                    CloudbreakEventService.DATAHUB_RESOURCE_TYPE, stackId);
         }
         return events;
     }
 
     @Override
-    public Page<StructuredNotificationEvent> cloudbreakEventsForStack(Long stackId, Pageable pageable) {
+    public Page<StructuredNotificationEvent> cloudbreakEventsForStack(Long stackId, String stackType, Pageable pageable) {
         return Optional.ofNullable(stackId)
-                .map(id -> structuredEventService.getEventsLimitedWithTypeAndResourceId(StructuredNotificationEvent.class, "stacks", id, pageable))
+                .map(id -> structuredEventService.getEventsLimitedWithTypeAndResourceId(StructuredNotificationEvent.class,
+                        getResourceType(stackType), id, pageable))
                 .orElse(Page.empty());
     }
 
+    private String getResourceType(String stackType) {
+        if ("WORKLOAD".equals(stackType)) {
+            return DATAHUB_RESOURCE_TYPE;
+        }
+        return stackType.toLowerCase();
```

Comment:
please return the constant DATALAKE_RESOURCE_TYPE value

Reply:
I'm not sure with the else brach. Because the datahub is different than the stack type I created a converter any other case I assume that is the same name.